src/core/db/backend/pg.py

```diff
@@ -154,12 +154,12 @@ def delete_repo(self, repo, force=False):
         res = self.execute_sql(query, params)
         return res['status']
 
-    def add_collaborator(self, repo, collaborator, privileges=[]):
+    def add_collaborator(self, repo, collaborator, db_privileges=[]):
```

Comment:
A failed call to add_collaborators() breaks further calls until the connection is reset or the server is restarted. If you try adding a user who doesn't exist or adding an invalid db privilege, you get "InternalError: current transaction is aborted, commands ignored until end of transaction block". Reproducible in the browser by trying to add a non-existent user as a collaborator.